src/mqtt.cpp

```diff
@@ -507,6 +509,25 @@ mqtt_publish_schedule() {
   }
 }
 
+bool
+mqtt_publish_config() {
+  if(!config_mqtt_enabled() || !mqttclient.connected() || evse.getEvseState() == OPENEVSE_STATE_STARTING) {
+    return false;
+  }
+  const size_t capacity = JSON_OBJECT_SIZE(128) + 1024;
+  DynamicJsonDocument doc(capacity);
+  config_serialize(doc, true, false, true);
+  mqtt_publish_json(doc, "/config");
+
+  if(config_version() == initial_config_version()) {
```

Comment:
The constant was hidden in the cpp file, that's why. I had to expose it to do that.

Reply:
Hum I guess, but why can't this be moved to the MQTT connection function?

Reply:
I think the responsibility to define the initial value for the version_number is up to the config class, which also makes use of the constant to initialise the variable.

It cannot be moved to the connection function because if the mqtt layer is connected before evse is started, the mqtt_publish_config will return because of `evse.getEvseState() == OPENEVSE_STATE_STARTING`. So the initial publication of the config_version will be done from the mqtt loop, not at connection time, when evse will have started.

Reply:
cases:

1. evse == `OPENEVSE_STATE_STARTING` + mqtt connected => `mqtt_publish_config` returns false
2. evse != `OPENEVSE_STATE_STARTING` + mqtt connected => `mqtt_publish_config` publishes both `/config` and `/config_version` (initial publish for first version)

Then, MQTT loop

1. `localVersion` != `config_version` +  evse == `OPENEVSE_STATE_STARTING` => `mqtt_publish_config` returns false
2. `localVersion` != `config_version` +  evse != `OPENEVSE_STATE_STARTING`
   2.1. publish `/config`
   2.2. fF `config_version` == `INITIAL_CONFIG_VERSION`,  publish `/config_version`

Reply:
Ok, guess we will leave it for now, just a lot of code to manage an irrelevant topic....

Reply:
I agree. But I also think that the code is now better than before and also answers [your request](https://github.com/OpenEVSE/ESP32_WiFi_V4.x/pull/634#discussion_r1188926242) to centralise things.  Making the app config class now also dependant on the evse settings now require to think about what could be the evse state when we access those settings. In most of the app, we do not care. But at startup, we care.